src/main/java/com/mparticle/kits/IterableRequest.java

```diff
@@ -0,0 +1,109 @@
+package com.mparticle.kits;
+
+import android.os.AsyncTask;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/**
+ * Async task to handle sending data to the Iterable server
+ * Created by David Truong dt@iterable.com
+ */
+class IterableRequest extends AsyncTask<IterableDeeplinkApiRequest, Void, String> {
+    static final String TAG = "IterableRequest";
+
+    static final String LOCATION_HEADER_FIELD    = "Location";
+
+    static final int DEFAULT_TIMEOUT_MS = 1000;   //1 seconds
+    static final long RETRY_DELAY_MS = 2000;      //2 seconds
+    static final int MAX_RETRY_COUNT = 5;
+
+    int retryCount = 0;
+    IterableDeeplinkApiRequest iterableApiRequest;
+    boolean retryRequest;
+
+    /**
+     * Sends the given request to Iterable using a HttpUserConnection
+     * Reference - http://developer.android.com/reference/java/net/HttpURLConnection.html
+     * @param params
+     * @return
+     */
+    protected String doInBackground(IterableDeeplinkApiRequest... params) {
+        if (params != null && params.length > 0) {
+            iterableApiRequest = params[0];
+        }
+
+        if (retryCount > 2) {
+            try {
+                Thread.sleep(RETRY_DELAY_MS * retryCount);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+
+        String requestResult = null;
+        if (iterableApiRequest != null) {
+            URL url;
+            HttpURLConnection urlConnection = null;
+
+            try {
+                url = new URL(iterableApiRequest.resourcePath);
+                urlConnection = (HttpURLConnection) url.openConnection();
+                urlConnection.setReadTimeout(DEFAULT_TIMEOUT_MS);
+                urlConnection.setInstanceFollowRedirects(false);
+
+                int responseCode = urlConnection.getResponseCode();
+                if (responseCode >= 300 && responseCode < 400) {
```

Comment:
The purpose of this request is to make a track call to iterable and return the destination URL. When the iterable url (denoted by the regex) is hit we track an event on our end and return destination url. If the url doesn't return a 3xx code we assume that it not a valid iterable deeplink and will return the original URL. If you have any suggestions of how to do this better feel free to let me know.

Looks like I refactored out too much code and took out the 5xx retry logic. I'll get that added back in.